networkx/generators/geometric.py

```diff
@@ -784,3 +785,110 @@ def thresholded_random_geometric_graph(
     )
     G.add_edges_from(edges)
     return G
+
+
+@py_random_state(5)
+def S1_graph(n, beta, gamma, mean_degree, kappas=None, seed=None):
+    r"""Returns a $\mathbb{S}^1$ model.
+
+    The $\mathbb{S}^1$ model is the simplest among the class of geometric models [1].
+    The similarity space is a one dimensional sphere—a circle of radius R—where N nodes are
+    distributed with a fixed density, set to one without loss of generality, so that $N = 2\pi R$.
+    Each node is also given a hidden variable ``'kappa'`` proportional to its expected degree.
+    The connection probability between a node i and a node j takes the form of a gravity law
+
+    $p_{ij} = \frac{1}{1 + \left( \frac{d_{ij}}{\mu \kappa_i \kappa_j} \right)^\beta}$
+
+    where $d_{ij} = R\Delta\theta_{ij}$ is the arc length of the circle between
+    nodes i and j separated by an angular distance $\Delta\theta_{ij}$. Parameters $\mu$ and $\beta$
+    control the average degree and the clustering coefficient, respectively.
+
+    Parameters
+    ----------
+    n: int
+        Size of the network
+    beta: float
+        Inverse temperature
+    gamma: float
+        Exponent of powerlaw distribution of the hidden degrees
+    mean_degree: float
+        The mean degree of the network
+    kappas: list, optional
+        Values of hidden degrees for each node
```

Comment:
Maybe it makes sense for `kappas` to be a dictionary with node as the key and kappa as the value?

Reply:
For now, I wanted to keep the implementation simple, but if you think that using dict will be better, I can extend it.